SimulationAgent/Agent.cs

```diff
@@ -218,6 +218,7 @@ private async Task CheckForStopOrStartToSimulationAsync()
             if (this.simulation != null && this.simulation.Enabled && !this.simulation.ShouldBeRunning())
             {
                 this.simulation.Enabled = false;
+                this.simulation.Modified = DateTimeOffset.Now;
```

Comment:
this could be done when calling simulations.UpdateAsync() so we're sure it will always happen and callers don't need to worry

Reply:
btw, this should be `UctNow`

Reply:
:white_check_mark: